[SRVCOM-1949] Add default broker backing channel config settings (#50649)
openshift/openshift-docs: modules/serverless-broker-backing-channel-default.adoc

// Module included in the following assemblies:
//
//  * serverless/admin_guide/serverless-configuration.adoc

:_content-type: PROCEDURE
[id="serverless-broker-backing-channel-default_{context}"]
= Configuring the default broker backing channel

> If you are using a channel-based broker, you can set the default backing channel type for the broker to either `InMemoryChannel` or `KafkaChannel`.

Reviewer: ```suggestion
If you use a channel-based broker, you can set the default backing channel type for the broker to either `InMemoryChannel` or `KafkaChannel`.
```

> set the default backing channel type for the broker to either `InMemoryChannel` or `KafkaChannel`.

Why/when should I do this? Consider adding explanation or something for information scent if the audience won't be absolutely sure.

Author: The differences between the channel implementations are explained here: https://docs.openshift.com/container-platform/4.11/serverless/discover/serverless-channels.html#serverless-channels-implementations

Tbh there's no reason why we'd recommend someone to use this, since we'd prefer them to move to the Kafka broker entirely (we've been moving to recommending that in our docs), but it was requested from eng that we document it anyway, since it's one of the main configuration options for this type of broker. If a user is using a channel-based broker over the Kafka broker, it can be assumed that they would understand why they would want to configure this, because the use case would be using channels that aren't the Kafka channel.

.Prerequisites

* You have administrator permissions on {product-title}.
* You have installed the {ServerlessOperatorName} and Knative Eventing on your cluster.
* If you want to use Kafka channels as the default backing channel type, you must also install the `KnativeKafka` CR on your cluster.

.Procedure

> * Modify the `KnativeEventing` custom resource (CR) to add configuration details for the `config-br-default-channel` config map:

Reviewer: To confirm, there's no need to apply or await anything, nor is there an obvious verification step?

Author: We can add an apply step, sure.
There's no obvious verification step or wait afaik.

Author: Added `oc apply` step and requirement to install the `oc` CLI as a prerequisite. I guess we didn't add it before because you can also update this from the web console, but we don't doc / have a request for specific steps for that, so adding this instead.

+
[source,yaml]
----

> apiVersion: operator.knative.dev/v1beta1

Reviewer 2: We still have just `apiVersion: operator.knative.dev/v1alpha1` as of Serverless <=1.25

> apiVersion: operator.knative.dev/v1beta1

Reviewer 2: ```suggestion
apiVersion: operator.knative.dev/v1alpha1
```

kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  config: <1>
    config-br-default-channel:
      channel-template-spec: |
        apiVersion: messaging.knative.dev/v1beta1
        kind: KafkaChannel <2>
        spec:
          numPartitions: 6 <3>

>           replicationFactor: 1 <4>

Reviewer 3: nit: this can be whatever but I would prefer something safer as an example:
```suggestion
          replicationFactor: 3 <4>
```

Reviewer 4: `3` is also what we have (for the broker) as default, b/c its more real, than just `1`

----
<1> In `spec.config`, you can specify the config maps that you want to add modified configurations for.
<2> The default backing channel type configuration. In this example, the default channel implementation for the cluster is `KafkaChannel`.

> <3> The number of partitions for the default Kafka channel that backs the broker.
> <4> The replication factor for the default Kafka channel that backs the broker.

Reviewer 3: ```suggestion
<3> The number of partitions for the Kafka channel that backs the broker.
<4> The replication factor for the Kafka channel that backs the broker.
```
